src/basic/mkdir.c

```diff
@@ -116,8 +115,8 @@ int mkdir_parents_internal(const char *prefix, const char *path, mode_t mode, mk
                         continue;
 
                 r = _mkdir(t, mode);
-                if (r < 0 && errno != EEXIST)
-                        return -errno;
```

Comment:
hmm, i think the old code was right, _mkdir is after a function parameter, where libc mkdir might be passed which returns errors in errno

Reply:
See #7659. I think this might be related.